Initial implementation of editor screens

smoogipoo opened the pull request:
I considered using OsuScreen/Screen here but deemed those implementations weren't really tailored for this sort of thing. The reason for that is that this implementation doesn't need the ability to push/exit screens and it doesn't need the escape key handling or the specific input handling in general.

We can iterate on the transition later, but I want something to work with for now.

smoogipoo committed:
Add test case for the editor

smoogipoo committed:
Remove items from the editor menu bar

smoogipoo committed:
Expose a mode change event from EditorMenuBar

smoogipoo committed:
Restructure Editor construction

smoogipoo committed:
Add EditorScreen + screen changing functionality

smoogipoo committed:
modeContainer -> screenContainer

smoogipoo committed:
Make EditorScreen a Container

smoogipoo committed:
Mask the editor screens

smoogipoo committed:
Adjust transforms of EditorScreen

smoogipoo committed:
Implement base Compose screen

smoogipoo committed:
Remove unnecessary usings

smoogipoo committed:
Give EditorScreen a Beatmap

peppy commented on osu.Game/Screens/Edit/Menus/EditorMenuBar.cs:
```diff
@@ -11,11 +11,19 @@
+        /// <summary>
+        /// Invaoked when the selected mode has changed.
+        /// </summary>
+        public event Action<EditorScreenMode> ModeChanged;
```
might be worth using a bindable for this?

peppy commented on osu.Game/Screens/Edit/Screens/EditorScreen.cs:
```diff
@@ -0,0 +1,41 @@
+        {
+            base.LoadComplete();
+
+            this.ScaleTo(0.5f).FadeTo(0)
```
0.5 sounds a bit extreme.

peppy commented:
How are you testing this when there's only one EditorScreen present? Also where is EditorScreenMode defined o_O (and does it need to be?)

smoogipoo committed:
Implement design mode

smoogipoo committed:
Reduce harshness of scale for now

Though I don't feel like we should worry about this much just yet until we have actual designs and can see how it looks. It's very well possible that we use different transitions here...

smoogipoo committed:
Re-namespace EditorScreenMode

smoogipoo committed:
Use a bindable for the current screen in EditorMenuBar

Replaces the current Action.

peppy committed:
Merge branch 'master' into editor-screens

peppy committed:
Update framework

peppy committed:
Remove scale effect on editor screen switches

smoogipoo committed:
Remove EditorMenuBarItem

smoogipoo committed:
Add File -> Exit to editor menu

smoogipoo committed:
Hook up BeatmapPanel "Edit" item

smoogipoo committed:
Remove rounded corners on the editor menu bar

peppy commented on osu.Game/Screens/Edit/Screens/EditorScreenMode.cs:
```diff
@@ -0,0 +1,19 @@
+        Design,
+        [Description("timing")]
+        Timing,
+        [Description("song")]
```
this one should come first, and probably be called "setup" not "song"

smoogipoo committed:
Fix bindable binding to make the editor load a screen by default again

smoogipoo committed:
Reorder screen tab control items

smoogipoo committed:
Make compose mode the default

Hopefully we can keep this at a ScreenSelectionTabControl level, but it may need to be moved to the Editor at some point. I'm leaving that for a future change however, if it's needed.

smoogipoo committed:
Put screens below the top and bottom bars of the editor
